util/smaractconfig.py

```diff
@@ -0,0 +1,160 @@
+        "axes": { str(c): {"range": [-1, 1], "unit": "m", "channel": c} for c in range(0, NUM_CHANNELS)
+        },
+}
+
+
+def calibrate(dev, channels):
+    if not channels:
+        return
+
+    """
+    Set calibration option bits to calibration_options. See section 2.7.1 in programmer manual
+    0 is for a default calibration
+    The following flags can be used:
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_DIRECTION
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_DIST_CODE_INV_DETECT
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_ASC_CALIBRATION
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_REF_MARK_TEST
+    smaract.SA_CTLDLL.SA_CTL_CALIB_OPT_BIT_LIMITED_TRAVEL_RANGE
+    """
+    calibration_options = 0
+
+    for channel in channels:
+        logging.info("Calibrating channel %d", channel)
+        # set calibration options
+        dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_CALIBRATION_OPTIONS,
+                        channel, calibration_options)
+
+        dev.Calibrate(channel)
+
+
+def autozero(dev, channels):
+    if not channels:
+        return
+
+    # Set auto-zero mode
+    for channel in channels:
+        logging.info("Setting auto-zero mode for channel %d", channel)
+        dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
+                            channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_AUTO_ZERO)  # auto zero
+
+    # Run the referencing
+    logging.info("Starting reference...")
+    dev.reference().result()
+    time.sleep(0.1)
+    logging.info("Complete")
+
+    # Set normal referencing mode
+    for channel in  channels:
+        logging.info("Setting the reference mode to normal for channel %d", channel)
+        dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
+                            channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_NORMAL)  # normal
+
+
+def set_zero(dev, channels):
+    """
+    Using the scale offset, set the current position of the controller to 0.
+    """
+    if not channels:
+        return
+
+    logging.info("Setting scale offsets. Current position: %s", dev.position.value)
+
+    for channel in channels:
+        logging.info("Setting the scale offset for channel %d", channel)
+        scale_offset = dev.GetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_LOGICAL_SCALE_OFFSET, channel)
+        position = dev.GetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_POSITION, channel)
+        # set the new scale offset
+        dev.SetProperty_i64(smaract.SA_CTLDLL.SA_CTL_PKEY_LOGICAL_SCALE_OFFSET, channel, scale_offset + position)
+
+    dev._updatePosition()
+    logging.info("New position: %s", dev.position.value)
+    logging.info("Complete.")
+
+
+def main(args):
+
+    parser = argparse.ArgumentParser(prog="smaractconfig",
+                                     description='Run a one time config for the SmarAct ')
+
+    parser.add_argument('--locator', required=True, type=str, help='Run the operations with the simulator', action="store")
```

Comment:
That help string is not really clear. Indicate how it should look like "network:sn:MCS2-00001601" and that "fake" can be used for a simulator.